### ISSUE TITLE
circular dependency numpy⇄mkl

### ISSUE THREAD

**Issue by scottdraves** [ContinuumIO/anaconda-issues#9117]
this wreaks havoc on anything that depends on numpy and uses a post-link script:
https://github.com/conda/conda/issues/7115
it seems to have appeared in the past month.

**Comment by scottdraves** [comment]
@kalefranz is this the right place for this issue?  i could not find an issue about this already filed.

**Comment by msarahan** [comment]
Yes, I think this is the right place.  I introduced that circular dependency - my apologies.  I'm not sure how to fix this - numpy is a build-time requirement of mkl_fft and mkl_random at the very least, and really also a runtime requirement for them to operate.  They're not strictly requirements for numpy, but we'd really like them to be installed by default with numpy because the speedups can be enormous.  https://www.anaconda.com/blog/developer-blog/improved-security-performance-in-anaconda-distribution-5/

**Comment by scottdraves** [comment]
I see -- how about splitting numpy into two packages, numpy-internal and numpy.
all the current contents of numpy would go into numpy-internal.
mkl would depend on numpy-internal.
numpy would depend on mkl and numpy-internal.
that breaks the cycle and still makes "conda install numpy" give the mkl accelerated version.

**Comment by scottdraves** [comment]
Also as a workaround, could we depend on an older version of numpy that doesn't depend on mkl?
what version would that be?

**Comment by msarahan** [comment]
Anaconda's recipes come from https://github.com/anacondarecipes .  See https://github.com/AnacondaRecipes/numpy-feedstock/blob/master/recipe/meta.yaml#L61

Numpy 1.13.3 build 3 or less, and numpy 1.14.2 build 0 do not have this change.

**Comment by scottdraves** [comment]
aha that makes a lot more sense, thanks will give that a shot.
